source/FreeRTOS_IP.c

```diff
@@ -1441,84 +1441,106 @@ BaseType_t xSendEventStructToIPTask( const IPStackEvent_t * pxEvent,
         /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
         /* coverity[misra_c_2012_rule_11_3_violation] */
         pxEthernetHeader = ( ( const EthernetHeader_t * ) pucEthernetBuffer );
 
-        /* Examine the destination MAC from the Ethernet header to see if it matches
-         * that of an end point managed by FreeRTOS+TCP. */
-        pxEndPoint = FreeRTOS_FindEndPointOnMAC( &( pxEthernetHeader->xDestinationAddress ), NULL );
-
-        if( pxEndPoint != NULL )
+        switch( pxEthernetHeader->usFrameType )
         {
-            /* The packet was directed to this node - process it. */
-            eReturn = eProcessBuffer;
-        }
-        else if( memcmp( xBroadcastMACAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-        {
-            /* The packet was a broadcast - process it. */
-            eReturn = eProcessBuffer;
+            case ipARP_FRAME_TYPE:
+            case ipIPv4_FRAME_TYPE:
+                #if ipconfigIS_DISABLED( ipconfigUSE_IPv4 )
+                    eReturn = eReleaseBuffer;
+                #endif
+                break;
+
+            case ipIPv6_FRAME_TYPE:
+                #if ipconfigIS_DISABLED( ipconfigUSE_IPv6 )
+                    eReturn = eReleaseBuffer;
+                #endif
+                break;
+
+            default:
+                if( FreeRTOS_ntohs( pxEthernetHeader->usFrameType ) <= 0x600U )
+                {
+                    #if ipconfigIS_ENABLED( ipconfigFILTER_OUT_NON_ETHERNET_II_FRAMES )
+                        eReturn = eReleaseBuffer;
+                    #endif
+                }
+                else
+                {
+                    #if ipconfigIS_DISABLED( ipconfigPROCESS_CUSTOM_ETHERNET_FRAMES )
+                        eReturn = eReleaseBuffer;
+                    #endif
+                }
+                break;
         }
-        else
-        #if ( ( ipconfigUSE_LLMNR == 1 ) && ( ipconfigUSE_DNS != 0 ) )
-            if( memcmp( xLLMNR_MacAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-            {
-                /* The packet is a request for LLMNR - process it. */
-                eReturn = eProcessBuffer;
-            }
-            else
-        #endif /* ipconfigUSE_LLMNR */
-        #if ( ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_DNS != 0 ) )
-            if( memcmp( xMDNS_MacAddress.ucBytes, pxEthernetHeader->xDestinationAddress.ucBytes, sizeof( MACAddress_t ) ) == 0 )
-            {
-                /* The packet is a request for MDNS - process it. */
-                eReturn = eProcessBuffer;
-            }
-            else
-        #endif /* ipconfigUSE_MDNS */
-        if( ( pxEthernetHeader->xDestinationAddress.ucBytes[ 0 ] == ipMULTICAST_MAC_ADDRESS_IPv6_0 ) &&
-            ( pxEthernetHeader->xDestinationAddress.ucBytes[ 1 ] == ipMULTICAST_MAC_ADDRESS_IPv6_1 ) )
```

Comment:
Here, all IPv6 multicast was allowed. Won't your change break existing IPv6 code? What about the solicited-node multicast addresses and the all nodes/routers multicast addresses?

I assume that this multicast code will change later, when we get the [MAC address filtering](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1065).

Reply:
Yeah, that part should probably be ignored until #1065. I suppose it should have released the buffer if ipconfigUSE_LLMNR or ipconfigUSE_MDNS were disabled and it matched xLLMNR_MacAddressIPv6 or xMDNS_MACAddressIPv6, then accept remaining multicast addresses.